### PR TITLE
add support for extraction functions

### PR DESCRIPTION
**Pull request by sladiwala**
To perform case insensitive equality searches. 

### Reviewers
@varmalho @andy256 @TetrationAnalytics/adm-team 

### Subscribers
@gshashitetration

### CODE REVIEW

File: extraction_function.go [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package godruid
+
+type ExtractionFn interface{}
```
Comment (varmalho) [inline extraction_function.go]: can we not make it more specific than just an interface{}?

Comment (ghost) [reply extraction_function.go]: at the very least it needs a `GetType() string`. "lower" for example. most extraction functions take arguments, but that one doesn't.

Comment (sladiwala) [reply extraction_function.go]: changed it to a struct with a `Type` field

**Comment by sladiwala** [comment]
Diffs have been updated. PTAL.
cc: @varmalho @andy256

**Comment by varmalho** [comment]
I don't understand this struct -- can you write some example pseudocode showing how client will pass this example extraction function -- I was suggesting of making the signature of extraction function more specific, i.e. if output is string etc.

**Comment by sladiwala** [comment]
@varmalho https://github.com/TetrationAnalytics/tetration/pull/27191 is how I was planning on using the extractionFn.

**Review by varmalho — APPROVED** [review]
lgtm

**Comment by varmalho** [comment]
got it, thanks!

**Comment by varmalho** [comment]
If you could reference this URL somewhere in the code that would be great

https://druid.apache.org/docs/latest/querying/dimensionspecs.html#output-types

**Comment by sladiwala** [comment]
@varmalho done

**Pull request merged**